pkg/managerdriver/driver.go

```diff
@@ -606,7 +606,7 @@ func (d *Driver) getTrafficPolicyJSON(ingress *netv1.Ingress, modSet *ingressv1a
 		return nil, err
 	}
 
-	if modSet == nil {
```

Comment:
This was the actual bug. The `modSet` passed was never nil, because we always return a non nil moduleset from:
https://github.com/ngrok/ngrok-operator/blob/7d62d40b77c532334390b0aa14eebcd36a01fea7/pkg/managerdriver/utils.go#L331-L338

Adding the `IsEmpty()` utility seemed like the easiest route here.